Library/MeterGeometry.cpp

```diff
@@ -0,0 +1,370 @@
+	std::map<LPCWSTR, std::pair<LPCWSTR, LPCWSTR>> postOptions;
+	if (!shapeOption.empty()) 
+		do
+		{
+			std::vector<std::wstring> shapeTokens = CustomTokenize(shapeOption.c_str(), L"|:");
+			m_Shapes.insert(std::pair<LPCWSTR, GeometryShape>(shapeName.c_str(), GeometryShape()));
+			bool usingMeasures = false;
+
+			std::wstring shapeType = L"", shapeParameters = L"";
+
+			for (int optionId = 0; optionId < shapeTokens.size(); optionId++)
+			{
+				if (shapeTokens.size() < optionId + 2) break; //not a pair
+
+				std::wstring optionName = shapeTokens[optionId].c_str();
+				std::wstring optionValue = shapeTokens[optionId + 1].c_str();
+
+				if (!m_Measures.empty())
+					usingMeasures = ContainsMeasures(optionValue);
+
+				//Store options to replace in update
+				if (!usingMeasures) {
+					if (!IsShape(optionName.c_str())) {
+						if (!IsPostOption(optionName.c_str())) {
+							if (!ReplaceShapeOption(m_Shapes[shapeName.c_str()], optionName.c_str(), optionValue.c_str()))
+								break;
+						}
+						else
+							postOptions[shapeOption.c_str()] = std::pair<LPCWSTR, LPCWSTR>(optionName.c_str(), optionValue.c_str());
+					}
+					else
+					{
+						
+						shapeType = optionName;
+						shapeParameters = optionValue;
+					}
+				}
+				else
+					m_MeasureOptions[shapeName.c_str()].push_back(std::pair<std::wstring, std::wstring>(optionName, optionValue));
+				++optionId;
+			}
+			if (shapeType.empty() || shapeParameters.empty())
+				break; //Invalid shape
+			if (!ParseShape(m_Shapes[shapeName.c_str()], shapeType.c_str(), shapeParameters.c_str()))
+				break;
+			shapeName = L"Shape" + std::to_wstring(++currentShapeId);
+			shapeOption = parser.ReadString(section, shapeName.c_str(), L"");
+		} while (!shapeOption.empty());
+
+	for (const auto& option : postOptions)
+	{
+		if (!ReplaceShapeOption(m_Shapes[option.first], option.second.first, option.second.second))
+			continue;
+	}
+	if (m_Initialized && m_Measures.empty() && !m_DynamicVariables)
+	{
+		Initialize();
+		m_NeedsRedraw = true;
+	}
+}
+
+bool MeterGeometry::Update()
+{
+	if (Meter::Update())
+	{
+		if (!m_Measures.empty() || m_DynamicVariables)
+		{
+			for (const auto& shape : m_MeasureOptions)
+			{
+				for (const auto& option : shape.second) {
+					std::wstring optionValue = option.second;
+					bool replaced = ReplaceMeasures(optionValue, AUTOSCALE::AUTOSCALE_OFF, 1.0, 6, false);
+					if (replaced)
+						if (!IsShape(option.first.c_str())) {
+							if (!ReplaceShapeOption(m_Shapes[shape.first], option.first.c_str(), optionValue.c_str()))
+								continue;
+						}
+						else
+							if (!ParseShape(m_Shapes[shape.first], option.first.c_str(), optionValue.c_str()))
+								continue;
+				}
+			}
+			return true;
+		}
+		else if (m_NeedsRedraw)
+		{
+			m_NeedsRedraw = false;
+			return true;
+		}
+	}
+	return false;
+}
+
+bool MeterGeometry::Draw(Gfx::Canvas & canvas)
+{
+	if (!Meter::Draw(canvas)) return false;
+	
+	for (const auto& shape : m_Shapes) {
```

Comment:
Can you change this to:
```
for (const auto& it : m_Shapes)
{
  const auto& shape = it.second;
}
```
and then use `shape` instead of `shape.second` below.